src/caffe/test/test_upgrade_proto.cpp

```diff
@@ -2889,6 +2889,7 @@ TEST_F(NetUpgradeTest, TestImageNet) {
   this->RunV1UpgradeTest(expected_v1_proto, expected_v2_proto);
 }  // NOLINT(readability/fn_size)
 
+#if defined (USE_OPENCV) && defined(USE_HDF5)
```

Comment:
Although ugly, it would be safer to always run this for the layers that are instantiated and have guards within the loop to skip the disabled layer types.